scarb/src/manifest_editor/add.rs

```diff
@@ -128,6 +130,17 @@ impl Dep {
                 DefaultBranch
             };
 
+            let git = match CanonicalUrl::new(&Url::parse(&git)?) {
+                Ok(url) => url.as_str().to_string(),
+                Err(_) => {
+                    ctx.opts
+                        .config
+                        .ui()
+                        .warn(format!("dependency ({name}) has an invalid git URL {git}"));
```

Comment:
Why not simply print error message here (without causes)? That would actually tell more what's making Scarb upset.

Reply:
i was thinking bcs on non `dry-run`, when its trying to resolve the workspace it will output the same error mesage so i was thinking maybe it'd be redundant?

Reply:
Oh that's true! It looks like this code shouldn't print anything at all then, and instead the verification step should do everything.